packages/amplify_core/lib/src/types/exception/missing_auth_exception.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_core.missing_auth_exception}
+/// Exception thrown when auth is missing.
+/// {@endtemplate}
+class MissingAuthException extends StorageException {
```

Comment:
Could be more specific `MissingAuthProviderException`?

Reply:
I think Dillon has a good point - we should just throw ConfigurationError instead.